python/lsst/afw/geom/spherePoint.py

```diff
@@ -0,0 +1,64 @@
+# // -*- lsst-c++ -*-
+
+# /*
+#  * LSST Data Management System
+#  * See COPYRIGHT file at the top of the source tree.
+#  *
+#  * This product includes software developed by the
+#  * LSST Project (http://www.lsst.org/).
+#  *
+#  * This program is free software: you can redistribute it and/or modify
+#  * it under the terms of the GNU General Public License as published by
+#  * the Free Software Foundation, either version 3 of the License, or
+#  * (at your option) any later version.
+#  *
+#  * This program is distributed in the hope that it will be useful,
+#  * but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+#  * GNU General Public License for more details.
+#  *
+#  * You should have received a copy of the LSST License Statement and
+#  * the GNU General Public License along with this program. If not,
+#  * see <http://www.lsstcorp.org/LegalNotices/>.
+#  */
+
+from future.utils import raise_from
+
+import lsst.pex.exceptions
+from ._spherePoint import SpherePoint
+
+__all__ = []  # import this module only for its side effects
+
+
+def addSpherePointMethods(cls):
+    def __getitem__(self, index):
+        if index < -len(self) or index > 1:
+            raise IndexError("Invalid index: %d" % index)
+        try:
+            if index >= 0:
+                return self._getitem(index)
+            else:
+                return self._getitem(len(self) + index)
+        except lsst.pex.exceptions.OutOfRangeError as e:
+            raise_from(IndexError(), e)
+    cls.__getitem__ = __getitem__
```

Comment:
Yes. I wrote one for STL collections but it takes the collection as an argument instead of the size. I should rewrite it to take the size and use it here.